prow/cmd/deck/job_history.go

```diff
@@ -240,16 +286,45 @@ func (bucket gcsBucket) listBuildIDs(root string) ([]int64, error) {
 	return ids, nil
 }
 
-func parseJobHistURL(url *url.URL) (bucketName, root string, buildID int64, err error) {
+// parseJobHistURL parses the job History URL
+// example urls:
+// * new format: https://prow.k8s.io/job-history/s3/kubernetes-jenkins/pr-logs/directory/pull-capi?buildId=1245584383100850177
+// * old format: https://prow.k8s.io/job-history/kubernetes-jenkins/pr-logs/directory/pull-capi?buildId=1245584383100850177
+// Newly generated URLs will include the storageType. We still support old URLs so they don't break.
+// For old URLs we assume that the storageType is `gs`.
+// examples return values:
+// * storageType: s3, gs
+// * bucketName: kubernetes-jenkins
+// * root: pr-logs/directory/pull-capi
+// * buildID: 1245584383100850177
+func parseJobHistURL(url *url.URL) (storageType, bucketName, root string, buildID int64, err error) {
 	buildID = emptyID
 	p := strings.TrimPrefix(url.Path, "/job-history/")
-	s := strings.SplitN(p, "/", 2)
-	if len(s) < 2 {
-		err = fmt.Errorf("invalid path (expected /job-history/<gcs-path>): %v", url.Path)
-		return
+	// examples for p:
+	// * new format: s3/prow-artifacts/pr-logs/directory/bazel-build-test
+	// * old format: kubernetes-jenkins/pr-logs/directory/pull-cluster-api-provider-openstack-test
+	if strings.HasPrefix(p, "gs") || strings.HasPrefix(p, "s3") {
+		// handle new format
+		s := strings.SplitN(p, "/", 3)
+		if len(s) < 3 {
+			err = fmt.Errorf("invalid path (expected /job-history/<storage-type>/<storage-path>): %v", url.Path)
+			return
+		}
+		storageType = s[0]
+		bucketName = s[1]
+		root = s[2] // `root` is the root "directory" prefix for this job's results
+	} else {
+		// handle old format
+		s := strings.SplitN(p, "/", 2)
+		if len(s) < 2 {
+			err = fmt.Errorf("invalid path (expected /job-history/<gcs-path>): %v", url.Path)
+			return
+		}
+		storageType = "gs"
```

Comment:
Another nit: All of the `gs` or `s3` should be a const that is define in one central place (pkg/io/providers?)

Reply:
Fixed